src/api/roles.coffee

```diff
@@ -15,7 +15,7 @@ utils = require '../utils'
 
 
 filterRolesFromChannels = (channels, allClients) ->
-  roles = {}
+  rolesMap = {} # K: permission, V: channels that share permission
```

Comment:
I think this will only return roles that have channels associated with them. Wouldn't we need to search clients as well. Especially since this is used to get all roles in line 49.